server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -771,6 +808,7 @@ protected Query getRegexpQuery(String field, String termStr) throws ParseExcepti
         for (Map.Entry<String, Float> entry : fields.entrySet()) {
             Query q = getRegexpQuerySingle(entry.getKey(), termStr);
             assert q != null;
+            onDiscoveredField(entry.getKey());
```

Comment:
Not needed, `getRegexpQuerySingle` discovers fields

Reply:
done